### PR TITLE
Unsupported theme single product enhancements

### PR DESCRIPTION
**Pull request by claudiulodro**
Related #17551 
(Notice this PR merges to a feature branch)

We're not going to be able to use the review comments template for unsupported themes. There were just too many themes that had issues rendering it properly. We'll have to use the theme's regular comment template. Everything else looks good and works well.

To test:
1. Use a theme that doesn't support WC (Bassist, Hamilton, Ginkaku, etc.)
2. Visit a single product page

EDIT: Noticed some code formatting tweaks I need to do, so don't merge this yet.

Screenshots (Chrome can now do full page screenshots!):
<details>

local wordpress dev-product-polo-
local wordpress dev-product-t-shirt-
</details>

### CODE REVIEW

**Comment by claudiulodro** [comment]
Once this is merged #17638 and this likely need to be integrated together so everything is done centrally.

New screenshots:

local wordpress dev-product-t-shirt- 3
local wordpress dev-product-v-neck-t-shirt- 1

File: assets/css/unsupported-theme.scss [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+ */
+
+.single-product {
+
+	// Hide duplicate product title.
+	h1.product_title {
+		display: none;
+	}
+
+	// Normalize the button and input to the same height and font size.
```
Comment (mikejolley) [inline assets/css/unsupported-theme.scss]: Let's try and roll these styles (this and below) into our default CSS if possible. Having dedicated CSS for unsupported themes seems weird to me?

File: assets/css/unsupported-theme.scss [hunk 2]
```diff
@@ -0,0 +1,31 @@
+/**
+ * unsupported-theme.scss
+ * Styles applied to themes that don't formally support WooCommerce to normalize elements.
+ */
+
+.single-product {
+
+	// Hide duplicate product title.
+	h1.product_title {
+		display: none;
```
Comment (mikejolley) [inline assets/css/unsupported-theme.scss]: If we can do this via the shortcode (arg)? That would be better than CSS IMO

File: assets/css/unsupported-theme.scss [hunk 3]
```diff
@@ -0,0 +1,31 @@
+/**
+ * unsupported-theme.scss
```
Comment (mikejolley) [inline assets/css/unsupported-theme.scss]: If we can address other CSS file feedback, let's drop it.

File: includes/theme-support/class-wc-unsupported-theme-product-single.php [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+}
+
+/**
+ * Single product pages support for themes that don't declare WooCommerce support.
+ *
+ * @class   WC_Unsupported_Theme_Product_Single
+ * @package WooCommerce/Classes
+ * @since   3.3.0
+ */
+class WC_Unsupported_Theme_Product_Single {
```
Comment (mikejolley) [inline includes/theme-support/class-wc-unsupported-theme-product-single.php]: Thinking about this in relation to my archive PR; do we have a separate class for each, a class combined for both, or use the template loader like I did?

File: includes/wc-template-functions.php [hunk 5]
```diff
@@ -2738,3 +2738,17 @@ function wc_page_noindex() {
 function wc_get_theme_slug_for_templates() {
 	return apply_filters( 'woocommerce_theme_slug_for_templates', get_option( 'template' ) );
 }
+
+/**
+ * Load up unsupported theme enhancements if needed.
+ *
+ * @todo Move this into the main unsupported theme loader.
+ */
+function wc_enhance_unsupported_theme_product_singles( $template ) {
```
Comment (mikejolley) [inline includes/wc-template-functions.php]: This function can probably be a method in the template loader class. It's all related IMO.

File: includes/theme-support/class-wc-unsupported-theme-product-single.php [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+		wp_enqueue_style( 'woocommerce-unsupported-theme' );
+	}
+
+	/**
+	 * Replace the content with the product shortcode.
+	 *
+	 * @param string $content Post content.
+	 * @return string
+	 */
+	public static function inject_product( $content ) {
```
Comment (mikejolley) [inline includes/theme-support/class-wc-unsupported-theme-product-single.php]: Worried this is too broad. Should be check we're inside the main query?

File: includes/theme-support/class-wc-unsupported-theme-product-single.php [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+	/**
+	 * Theme enhancements init.
+	 */
+	public static function init() {
+		remove_action( 'woocommerce_before_main_content', 'woocommerce_output_content_wrapper', 10 );
+		remove_action( 'woocommerce_after_main_content', 'woocommerce_output_content_wrapper_end', 10 );
+		remove_filter( 'template_include', array( 'WC_Template_Loader', 'template_loader' ) );
+		remove_filter( 'comments_template', array( 'WC_Template_Loader', 'comments_template_loader' ) );
+
+		add_theme_support( 'wc-product-gallery-zoom' );
```
Comment (mikejolley) [inline includes/theme-support/class-wc-unsupported-theme-product-single.php]: If we're enabling by default we need to ensure we add this inside our new theming docs.

**Review by mikejolley — CHANGES_REQUESTED** [review]
Few items of feedback mostly around CSS and how this meshes with other changes (archives).

File: includes/theme-support/class-wc-unsupported-theme-product-single.php (hunk 4 above)
Comment (claudiulodro) [reply includes/theme-support/class-wc-unsupported-theme-product-single.php]: I'm going to move it into the template loader. The whole class is really only a couple methods.

**Pull request merged**